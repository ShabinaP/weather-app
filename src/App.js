import React, { useState, useEffect } from "react";
import getForecast from "./requests/getForecast";
import LocationDetails from "./Components/LocationDetails";
import SearchForm from "./Components/SearchForm";
import ForecastSummaries from "./Components/ForecastSummaries";
import ForecastDetails from "./Components/ForecastDetails";
import "./styles/App.css";
import "./styles/index.css";

const App = () => {
  const [forecasts, setForecasts] = useState([]);
  const [location, setLocation] = useState({ city: "", country: "" });
  const [searchText, setSearchText] = useState("");
  const [selectedDate, setSelectedDate] = useState(0);
  const [errorMessage, setErrorMessage] = useState("");
  const selectedForecast = forecasts.find(
    (forecast) => forecast.date === selectedDate
  );
  const handleForecastSelect = (date) => {
    setSelectedDate(date);
  };
  const handleCitySearch = () => {
    getForecast(
      searchText,
      setSelectedDate,
      setForecasts,
      setLocation,
      setErrorMessage
    );
  };

  useEffect(() => {
    getForecast(
      searchText,
      setSelectedDate,
      setForecasts,
      setLocation,
      setErrorMessage
    );
  }, []);
  return (
    <div className="weather-app">
      <LocationDetails
        city={location.city}
        country={location.country}
        errorMessage={errorMessage}
      />
      <SearchForm
        onSubmit={handleCitySearch}
        searchText={searchText}
        setSearchText={setSearchText}
      />
      {!errorMessage && (
        <>
          <ForecastSummaries
            forecasts={forecasts}
            onForecastSelect={handleForecastSelect}
          />
          {selectedForecast && <ForecastDetails forecast={selectedForecast} />}
        </>
      )}
    </div>
  );
};

export default App;
