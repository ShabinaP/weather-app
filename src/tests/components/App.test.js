import { render } from "@testing-library/react";
import React from "react";
import App from "../../Components/App";

test("renders correctly", () => {
  const { asFragment } = render(<App />);

  expect(asFragment()).toMatchSnapshot();
});
