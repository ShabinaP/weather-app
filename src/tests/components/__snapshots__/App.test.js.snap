// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="weather-app"
  >
    <h1
      class="location-details"
    >
      , 
    </h1>
    <div
      class="search-form"
    >
      <form>
        <input
          name="search"
          placeholder="Search city..."
          type="text"
          value=""
        />
        <button
          class="searchButton"
          type="submit"
        >
          SEARCH
        </button>
      </form>
    </div>
    <div
      class="forecast-summaries"
    />
  </div>
</DocumentFragment>
`;
